src/components/question-editor/QuestionEditor.tsx

```diff
@@ -12,35 +12,60 @@ import FormGroup from '@mui/material/FormGroup'
 import InputBase from '@mui/material/InputBase'
 import Radio from '@mui/material/Radio'
 import RadioGroup from '@mui/material/RadioGroup'
+import Autocomplete from '@mui/material/Autocomplete'
+import TextField from '@mui/material/TextField'
+import Typography from '@mui/material/Typography'
 
 import AppTextField from '~/components/app-text-field/AppTextField'
 import AppSelect from '~/components/app-select/AppSelect'
+import useForm from '~/hooks/use-form'
+import AppButton from '~/components/app-button/AppButton'
+import { sortQuestions } from '~/components/question-editor/QuestionEditor.constants'
 
 import { styles } from '~/components/question-editor/QuestionEditor.styles'
-import { sortQuestions } from '~/components/question-editor/QuestionEditor.constants'
-import { Answer, SizeEnum, TextFieldVariantEnum } from '~/types'
+import {
+  ButtonTypeEnum,
+  ButtonVariantEnum,
+  ComponentEnum,
+  QuestionForm,
+  SizeEnum,
+  TextFieldVariantEnum,
+  TypographyVariantEnum
+} from '~/types'
 
-const QuestionEditor = () => {
-  const [questionType, setQuestionType] = useState(sortQuestions[0])
-  const [question, setQuestion] = useState<string>('')
-  const [answers, setAnswers] = useState<Answer[]>([])
-  const [answer, setAnswer] = useState<string>('')
+interface QuestionEditorProps {
+  loading: boolean
+  fetchData: (data?: QuestionForm) => Promise<void>
+}
 
+const QuestionEditor: FC<QuestionEditorProps> = ({ fetchData, loading }) => {
   const { t } = useTranslation()
+  const navigate = useNavigate()
 
-  const isMultipleChoice = questionType.value === sortQuestions[0].value
-  const isOpenAnswer = questionType.value === sortQuestions[1].value
-  const isSingleChoice = questionType.value === sortQuestions[2].value
-  const isEmptyAnswer = answers[answers.length - 1]?.text === ''
+  const { data, handleInputChange, handleNonInputValueChange, handleSubmit } =
+    useForm<QuestionForm>({
+      initialValues: {
+        type: sortQuestions[0].value,
+        title: '',
+        text: '',
+        answers: [],
+        openAnswer: ''
+      },
+      onSubmit: async () => {
+        await fetchData(data)
+      }
+    })
 
-  const setTypeValue = (value: string) => {
-    const questionOption = sortQuestions.find((item) => item.value === value)
+  const { type, title, text, answers, openAnswer } = data
 
-    setQuestionType(questionOption ?? sortQuestions[0])
-  }
+  const isMultipleChoice = type === sortQuestions[0].value
+  const isOpenAnswer = type === sortQuestions[1].value
+  const isSingleChoice = type === sortQuestions[2].value
```

Comment:
I guess we will use this in other places, like a question table, It may make sense to create a function which will accept a type and return { isMultipleChoice, isOpenAnswer, isOpenAnswer }

Reply:
Added function to constants